app/src/main/java/fragments/TempFeedFragment.java

```diff
@@ -206,7 +168,7 @@ public void onNothingSelected(AdapterView<?> parent) {
     }
 
     public void getWeatherDetails(String locationID){
-        String temp_url = url + "?id=" + locationID + "&appid=" + api_key;
+        String temp_url = API_URL + "?id=" + locationID + "&appid=" + API_KEY;
```

Comment:
I'd say the pretty version of this would look something like this:
WeatherUtil.requestWeatherForLocation(String locationID, new WeatherUtil.onWeatherResponse(Weather weather) {
sunsetTime = weather.getSunsetTime();
tvWeather.setText(weather.getDescription());
tvAirTemp.setText(weather.getTemperature());
}

And then you just move all of the JSON parsing into a Weather class constructor and url creation can be left in the WeatherUtils class.

This way your fragment just does "get the weather and when it comes back fill in these fields" and doesn't really have to care how it happens.